nixos/tests/nix-ld.nix

```diff
@@ -0,0 +1,20 @@
+import ./make-test-python.nix ({ lib, pkgs, ...} :
+{
+  name = "nix-ld";
+  nodes.machine = { pkgs, ... }: {
+    programs.nix-ld.enable = true;
+    environment.systemPackages = [
+      (pkgs.runCommand "patched-hello" {} ''
+        install -D -m755 ${pkgs.hello}/bin/hello $out/bin/hello
+        patchelf $out/bin/hello --set-interpreter ${pkgs.nix-ld.ldPath}
```

Comment:
Why do we need to patchelf the hello executable in this test, isn't nix-ld supposed to chainload behind `/lib64/ld-linux-x86-64.so.2` anyways?

Once this is answered, I'd LGTM this.

Reply:
The hello package uses the correct ld-linux-x86-64.so.2 from it's glibc. We need to patch it back to use the global `/lib64/ld-linux-x86-64.so.2` in this nixos test so we can test that nix-ld actually works as intended.